src/pds/registrysweepers/utils/db/__init__.py

```diff
@@ -22,59 +24,59 @@
 
 
 def query_registry_db(
-    host: Host,
+    client: OpenSearch,
     query: Dict,
     _source: Dict,
     index_name: str = "registry",
     page_size: int = 10000,
     scroll_keepalive_minutes: int = 10,
 ) -> Iterable[Dict]:
     """
-    Given an OpenSearch host and query/_source, return an iterable collection of hits
+    Given an OpenSearch client and query/_source, return an iterable collection of hits
 
-    Example query: {"bool": {"must": [{"terms": {"ops:Tracking_Meta/ops:archive_status": ["archived", "certified"]}}]}}
+    Example query: {"query: {"bool": {"must": [{"terms": {"ops:Tracking_Meta/ops:archive_status": ["archived", "certified"]}}]}}}
     Example _source: {"includes": ["lidvid"]}
     """
 
-    req_content = {
-        "query": query,
-        "_source": _source,
-        "size": page_size,
-    }
-
+    scroll_keepalive = f"{scroll_keepalive_minutes}m"
     query_id = get_random_hex_id()  # This is just used to differentiate queries during logging
-    log.info(f"Initiating query with id {query_id}: {req_content}")
-
-    path = f"{index_name}/_search?scroll={scroll_keepalive_minutes}m"
+    log.info(f"Initiating query with id {query_id}: {query}")
 
     served_hits = 0
 
     last_info_log_at_percentage = 0
     log.info(f"Query {query_id} progress: 0%")
 
     more_data_exists = True
+    scroll_id = None
     while more_data_exists:
-        resp = retry_call(
-            auto_raise_for_status(requests.get),
-            fargs=[urllib.parse.urljoin(host.url, path)],
-            fkwargs={"auth": (host.username, host.password), "verify": host.verify, "json": req_content},
-            exceptions=(HTTPError, RuntimeError),
+        if scroll_id is None:
+            fetch_func = lambda: client.search(  # noqa: E731
```

Comment:
See Slack discussion for assigning lambdas to variables

Reply:
resolved